handlers/google_apps_api_v1/handler/init.rb

```diff
@@ -0,0 +1,165 @@
+# Require the dependencies file to load the vendor libraries
+require File.expand_path(File.join(File.dirname(__FILE__), "dependencies"))
+
+class GoogleAppsApiV1
+  # ==== Parameters
+  # * +input+ - The String of Xml that was built by evaluating the node.xml handler template.
+  def initialize(input)
+    # Set the input document attribute
+    @input_document = REXML::Document.new(input)
+
+    # Retrieve all of the handler info values and store them in a hash variable named @info_values.
+    @info_values = {}
+    REXML::XPath.each(@input_document, "/handler/infos/info") do |item|
+      @info_values[item.attributes["name"]] = item.text.to_s.strip
+    end
+
+    # Retrieve all of the handler parameters and store them in a hash variable named @parameters.
+    @parameters = {}
+    REXML::XPath.each(@input_document, "/handler/parameters/parameter") do |item|
+      @parameters[item.attributes["name"]] = item.text.to_s.strip
+    end
+
+    @debug_logging_enabled = ["yes","true"].include?(@info_values['enable_debug_logging'].downcase)
+    @error_handling = @parameters["error_handling"]
+
+    @service_account = @info_values["service_account"]
+    @private_key = formatPrivateKey(@info_values["private_key"])
+
+    @body = @parameters["body"].nil? ? "" : @parameters["body"]
+    @method = (@parameters["method"] || :get).downcase.to_sym
+    @scope = @parameters["scope"]
+    @user = @parameters["user"]
+    @url = @parameters["url"]
+
+    @accept = :json
+    @content_type = :json
+  end
+
+  def execute
+    # Initialize return data
+    error_message = nil
+    error_key = nil
+    response_code = nil
+
+    begin
+      api_route = "#{@url}"
+      puts "API ROUTE: #{@method.to_s.upcase} #{api_route}" if @debug_logging_enabled
+      puts "BODY: \n #{@body}" if @debug_logging_enabled
+
+      # get assertion needed for token request
+      assertion = getAssertion()
+
+      # Make token request
+      tokenResponse = RestClient::Request.execute \
+        method: :post, \
+        url: "https://oauth2.googleapis.com/token", \
+        payload: {
+          grant_type: "urn:ietf:params:oauth:grant-type:jwt-bearer",
+          assertion: assertion
+        }, \
+        headers: {:content_type => "application/x-www-form-urlencoded", :accept => @accept}
+
+      token = JSON.parse(tokenResponse)["access_token"]
+      puts "Token successfully retrieved" if token
```

Comment:
if @debug_logging_enabled

Is this even necessary? Probably should print an error message instead if the token was not retrieved.